File: Atcoder_#326/javascript/c_peak.js
function main(input) {
    const [n, m, ...a] = input.trim().split(/\s+/).map(Number);

    // プレゼントの座標をソート
    a.sort((x, y) => x - y);

    let maxCount = 0;
    let j = 0;

    for (let i = 0; i < n; i++) {
        for (let k = j; k < n; k++) {
            if (a[k] >= a[i] + m) {
                break;
            }
            j = k + 1;
        }

        // 現在の範囲で取得できるプレゼントの数を計算
        maxCount = Math.max(maxCount, j - i);
    }

    console.log(maxCount);
}

// 標準入力を読み込み、main関数を呼び出し
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);

File: Atcoder_#326/ruby/c_peak.rb
n, m = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

# プレゼントの座標をソート
a.sort!

max_count = 0
j = 0

(0...n).each do |i|
  # j を進めて、a[j] が a[i] + M 以上になる直前まで調整
  j += 1 while j < n && a[j] < a[i] + m
  # 現在の範囲で取得できるプレゼントの数を計算
  max_count = [max_count, j - i].max
end

puts max_count
